src/passes/OptimizeCasts.cpp

```diff
@@ -453,20 +437,32 @@ struct BestCastFinder : public LinearExecutionWalker<BestCastFinder> {
   void visitRefCast(RefCast* curr) { handleRefinement(curr); }
 
   void handleRefinement(Expression* curr) {
+    auto* teeFallthrough = Properties::getFallthrough(
+      curr, options, *getModule(), Properties::FallthroughBehavior::NoTeeBrIf);
+    if (auto* set = teeFallthrough->dynCast<LocalSet>()) {
```

Comment:
```suggestion
    if (auto* tee = teeFallthrough->dynCast<LocalSet>()) {
```
Also the `isTee` check below is not needed, as a fallthrough LocalSet must be a tee (because only a tee flows out a value that can fall through to some place).